syncd/syncd_notifications.cpp

```diff
@@ -92,19 +92,122 @@ void redisPutFdbEntryToAsicView(
         return;
     }
 
-    if ((fdb->event_type == SAI_FDB_EVENT_AGED)||(fdb->event_type == SAI_FDB_EVENT_FLUSHED))
+    if (fdb->event_type == SAI_FDB_EVENT_AGED)
     {
-        if (fdb->event_type == SAI_FDB_EVENT_AGED)
+        SWSS_LOG_DEBUG("remove fdb entry %s for SAI_FDB_EVENT_AGED",key.c_str());
+        g_redisClient->del(key);
+        return;
+    }
+
+    if (fdb->event_type == SAI_FDB_EVENT_FLUSHED)
+    {
+        sai_vlan_id_t vlan_id = fdb->fdb_entry.vlan_id;
+        sai_object_id_t port_oid = 0;
+        int port_oid_found = 0;
+        int attr_num = fdb->attr_count;
+        
+        for (int i = 0; i < attr_num; i++)
         {
-            SWSS_LOG_DEBUG("remove fdb entry %s for SAI_FDB_EVENT_AGED",key.c_str());
+            if(fdb->attr[i].id == SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID)
+            {
+                port_oid = fdb->attr[i].value.oid;
+                port_oid_found = 1;
+            }
         }
-        if (fdb->event_type == SAI_FDB_EVENT_FLUSHED)
+        
+        if (!port_oid_found)
         {
-            SWSS_LOG_DEBUG("remove fdb entry %s for SAI_FDB_EVENT_FLUSHED",key.c_str());
+            SWSS_LOG_ERROR("Failed to get bridge port ID for FDB entry %s",strFdbEntry.c_str());
+            return;
         }
-        g_redisClient->del(key);
+                
+        if( !port_oid && !vlan_id )
+        {
+            /* we got a flush all fdb event here */
+            /* exaple of a flush all fdb event   */
+            /*
+            [{
+            "fdb_entry":"{
+                \"bridge_id\":\"oid:0x0\",
+                \"bridge_type\":\"SAI_FDB_ENTRY_BRIDGE_TYPE_1Q\",
+                \"mac\":\"00:00:00:00:00:00\",
+                \"switch_id\":\"oid:0x21000000000000\",
+                \"vlan\":\"0\"}",
+            "fdb_event":"SAI_FDB_EVENT_FLUSHED",
+                "list":[
+                    {"id":"SAI_FDB_ENTRY_ATTR_BRIDGE_PORT_ID","value":"oid:0x0"},
+                    {"id":"SAI_FDB_ENTRY_ATTR_TYPE","value":"SAI_FDB_ENTRY_TYPE_DYNAMIC"},
+                    {"id":"SAI_FDB_ENTRY_ATTR_PACKET_ACTION","value":"SAI_PACKET_ACTION_FORWARD"}
+                ]
+            }]
+            */
+            SWSS_LOG_NOTICE("received a flush all fdb event");
```

Comment:
this behaviour is not described in SAI headers that when no port and no vlan id then 1 notification is send to notify that all have been flushed from the device, please put that description (and consult with SAI community?) whether that behavior is good 
for example, 
-all entries were flushed if "mac == all 0, vlanid == 0, bridge_id == 0"

this is fine by me, but please update SAI headers

Reply:
I will talk to SAI guys see whether they already have something similar if not may ask them to add.